dangerzone/isolation_provider/container.py

```diff
@@ -170,9 +195,11 @@ def exec(
         ) as p:
             if p.stdout is not None:
                 for line in p.stdout:
-                    (error, text, percentage) = self.parse_progress(document, line)
-                    if stdout_callback:
-                        stdout_callback(error, text, percentage)
+                    progress_tuple = self.parse_progress(document, line)
```

Comment:
This is the first entry point for the (possibly) attacker-controlled string. I suggest removing/replacing control characters while before passing it to lower layers. This can be the basis for our sanitization logic when dealing with untrusted input. See here: https://stackoverflow.com/questions/4324790/removing-control-characters-from-a-string-in-python